crypto/rsa/rsa_ameth.c

```diff
@@ -34,7 +34,7 @@ static int rsa_param_encode(const EVP_PKEY *pkey,
 
     *pstr = NULL;
     /* If RSA it's just NULL type */
-    if (pkey->ameth->pkey_id == EVP_PKEY_RSA) {
+    if (pkey->ameth->pkey_id != EVP_PKEY_RSA_PSS) {
```

Comment:
I wonder if it makes sense to use `pkey_id == EVP_PKEY_RSA || pkey_id == EVP_PKEY_RSA2`. I assume that at some point we'll add also add OAEP, but I guess we can worry about that when we add it.

Reply:
Yes, as a side-effect this should allow a workaround for rsaOaep (without actually
implementing the correct Oaep parameter structure, see #7907)